### ISSUE TITLE
Issue during installation

### ISSUE THREAD

**Issue by piotr-yuxuan** [ebbywiselyn/emacs-kafka#2]
Hi! First of all, thanks a lot for this package. The screenshot is appealing, I'm very interested in trying it :-) However I'm stumbling upon something unexpected during installation:

```
Error (use-package): kafka-cli/:catch: Malformed arglist: (message not implemented)
```

My `init.el` contains:
``` elisp
(use-package kafka-cli
  :after magit-popup
  :straight (emacs-kafka :host github :repo "ebbywiselyn/emacs-kafka"))
```

Versions I use:
-  `magit-popup "ab75385a1fb8c0fba0769d448b13ba8324835261"`
- `use-package "05a4033b830bf52c596ceedea10b2cbd91f85fdb"`
- `straight.el "ac0ca487eccc102e58f2531252356404dfe7df0e"` (reproductible package management)

Just tell me if I can provide you any other useful information.

**Comment by ebbywiselyn** [comment]
Hello,

Thanks for your kind words and for reporting the issue.

Sorry for responding so late I been away for a while.

I haven't used `straight` before so I tried bootstrapping the latest version and the following seem to work.

Bootstrap script
```
(let ((bootstrap-file (concat user-emacs-directory "straight/repos/straight.el/bootstrap.el"))
      (bootstrap-version 3))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
	(url-retrieve-synchronously
	 "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
	 'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))
```



```
(straight-use-package `(emacs-kafka :type git :host github
				    :repo "ebbywiselyn/emacs-kafka"
				    :branch "master"))
(setq kafka-cli-bin-path "/path/to/apps/kafka_2.10-0.10.2.1/bin")
(setq kafka-cli-config-path "/path/to/apps/kafka_2.10-0.10.2.1/config")

M-x run-zookeeper
M-x run-kafkabroker
M-x run-kafkaconsumer
				    
```
Just remember to download the kafka binaries and use the correct path above.

Could you let me know if this works ?

**Comment by piotr-yuxuan** [comment]
Thanks for your reply. 

#### TL;DR

Are variables `kafka-cli-bin-path` and `kafka-cli-config-path` required at package loading time? If not, my guess is that some more package is needed. Would you mind give me the list of loaded packages (and their versions) when `emacs-kafka` is being loaded on your machine?

#### Detailed answer

Here is my full minimal `init.el` when I want to install `emacs-kafka`:

``` elisp
(let ((bootstrap-file (concat user-emacs-directory "straight/repos/straight.el/bootstrap.el"))
      (bootstrap-version 3))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max)) 
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

;; Straight.el integration so `use-package` is equal to `straight-use-package`
(straight-use-package 'use-package)
(setq-default straight-use-package-by-default t)

(use-package magit)
(use-package magit-popup)
(use-package kafka-cli
  :after magit-popup
  :custom (;; These custom variables will be needed once the package is loaded without errors.
	   (kafka-cli-bin-path "/tmp")
	   (kafka-cli-config-path "/tmp"))
  :straight (emacs-kafka :host github :repo "ebbywiselyn/emacs-kafka"))
```

I run `emacs -Q init.el` then I eval each form one after one. On the last one, I get this error:

``` elisp
Debugger entered--Lisp error: (error "Malformed arglist: (message not implemented)")
  signal(error ("Malformed arglist: (message not implemented)"))
  error("Malformed arglist: %s" (message "not implemented"))
  #f(compiled-function (name arglist &optional docstring decl &rest body) "Define NAME as a function.\nThe definition is (lambda ARGLIST [DOCSTRING] BODY...).\nSee also the function `interactive'.\nDECL is a declaration, optional, of the form (declare DECLS...) where\nDECLS is a list of elements of the form (PROP . VALUES).  These are\ninterpreted according to `defun-declarations-alist'.\nThe return value is undefined." #<bytecode 0x40081ce7>)(do-nothing (message "not implemented"))
  (defun do-nothing (message "not implemented"))
  eval-buffer(#<buffer  *load*> nil "/Users/piotr-yuxuan/.emacs.d/straight/build/emacs-kafka/kafka-cli.el" nil t)  ; Reading at buffer position 6741
  load-with-code-conversion("/Users/piotr-yuxuan/.emacs.d/straight/build/emacs-kafka/kafka-cli.el" "/Users/piotr-yuxuan/.emacs.d/straight/build/emacs-kafka/kafka-cli.el" t t)
  require(kafka-cli nil t)
  (not (require 'kafka-cli nil t))
  (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error))
  (progn (customize-set-variable 'kafka-cli-bin-path "/tmp" "Customized with use-package kafka-cli") (customize-set-variable 'kafka-cli-config-path "/tmp" "Customized with use-package kafka-cli") (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error)))
  (lambda nil (progn (customize-set-variable 'kafka-cli-bin-path "/tmp" "Customized with use-package kafka-cli") (customize-set-variable 'kafka-cli-config-path "/tmp" "Customized with use-package kafka-cli") (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error))))()
  eval-after-load(magit-popup (lambda nil (progn (customize-set-variable 'kafka-cli-bin-path "/tmp" "Customized with use-package kafka-cli") (customize-set-variable 'kafka-cli-config-path "/tmp" "Customized with use-package kafka-cli") (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error)))))
  (condition-case err (eval-after-load 'magit-popup (function (lambda nil (progn (customize-set-variable 'kafka-cli-bin-path "/tmp" "Customized with use-package kafka-cli") (customize-set-variable 'kafka-cli-config-path "/tmp" "Customized with use-package kafka-cli") (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error)))))) ((debug error) (funcall use-package--warning5 :catch err)))
  (progn (straight-use-package '(emacs-kafka :host github :repo "ebbywiselyn/emacs-kafka")) (defvar use-package--warning5 (function (lambda (keyword err) (let ((msg (format "%s/%s: %s" 'kafka-cli keyword (error-message-string err)))) (display-warning 'use-package msg :error))))) (condition-case err (eval-after-load 'magit-popup (function (lambda nil (progn (customize-set-variable 'kafka-cli-bin-path "/tmp" "Customized with use-package kafka-cli") (customize-set-variable 'kafka-cli-config-path "/tmp" "Customized with use-package kafka-cli") (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error)))))) ((debug error) (funcall use-package--warning5 :catch err))))
  eval((progn (straight-use-package '(emacs-kafka :host github :repo "ebbywiselyn/emacs-kafka")) (defvar use-package--warning5 (function (lambda (keyword err) (let ((msg (format "%s/%s: %s" 'kafka-cli keyword (error-message-string err)))) (display-warning 'use-package msg :error))))) (condition-case err (eval-after-load 'magit-popup (function (lambda nil (progn (customize-set-variable 'kafka-cli-bin-path "/tmp" "Customized with use-package kafka-cli") (customize-set-variable 'kafka-cli-config-path "/tmp" "Customized with use-package kafka-cli") (if (not (require 'kafka-cli nil t)) (display-warning 'use-package (format "Cannot load %s" 'kafka-cli) :error)))))) ((debug error) (funcall use-package--warning5 :catch err)))) nil)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

Just to make it sure, I don't use `package.el` at all. It means that no packages are loaded expect those built into Emacs: variable `package-activated-list` is set to `nil`. The only loaded packages from `init.el` are: `magit`, `magit-popup`, `kafka-cli`, and their dependencies. Here are the versions of them:

``` elisp
(("dash.el" . "c1991d4c22f356be21df6b3badd7233a94df7937")
 ("emacs-async" . "15bcbf6beae65d7606f0655711159ca56f050c6b")
 ("emacs-kafka" . "5b93e16a26d5f56fbdd7338024c2702f8e26e642")
 ("epkgs" . "eff485cb661c9f7d276043cf1b0b38b598fb4303")
 ("ghub" . "f1b647faf5ce5f033728236b9263e7ecee8f536f")
 ("magit" . "91b4af273e96fe02ce86c83a3a6071ea043edb1b")
 ("magit-popup" . "ab75385a1fb8c0fba0769d448b13ba8324835261")
 ("melpa" . "c9ceaeee2c87cd5d954e57c31ec26a6aee19346f")
 ("straight.el" . "ac0ca487eccc102e58f2531252356404dfe7df0e")
 ("use-package" . "05a4033b830bf52c596ceedea10b2cbd91f85fdb")
 ("with-editor" . "04d59d68dab58a7cf3034c84d8ba0553b78ae30c"))
```

**Comment by piotr-yuxuan** [comment]
Aw, got it! I'm submitting a patch! This will be my first one ever in elisp 🎉
